python/rmm/_lib/device_buffer.pyx

```diff
@@ -133,6 +133,27 @@ cdef class DeviceBuffer:
         }
         return intf
 
+    def copy(self):
+        """Returns a deep copy of DeviceBuffer
+
+        Returns
+        -------
+        A deep copy of existing ``DeviceBuffer``
+
+        Examples
+        --------
+        >>> import rmm
+        >>> db = rmm.DeviceBuffer.to_device(b"abc")
+        >>> db_copy = db.copy()
+        >>> db.copy_to_host()
+        array([97, 98, 99], dtype=uint8)
+        >>> db_copy.copy_to_host()
+        array([97, 98, 99], dtype=uint8)
```

Comment:
I'd like to see a piece of the example that shows that the copy is deep.
```suggestion
        array([97, 98, 99], dtype=uint8)
        >>> assert db is not db_copy
        >>> assert db.ptr != db_copy.ptr
```

Reply:
Updated 👍